lib/exq_scheduler/scheduler/server.ex

```diff
@@ -83,41 +80,44 @@ defmodule ExqScheduler.Scheduler.Server do
   end
 
   def handle_info(:tick, state) do
-    timeout =
-      if Storage.storage_connected?(state.storage_opts) do
-        now = Time.now()
-        handle_tick(state, now)
+    timeout = handle_tick(state)
+    next_tick(self(), timeout)
+    {:noreply, state}
+  end
 
-        sch_time = nearest_schedule_time(state, now)
+  defp handle_tick(state) do
+    if Storage.storage_connected?(state.storage_opts) do
+      active_schedules = Storage.active_schedules(state.storage_opts, state.schedules)
+
+      if !Enum.empty?(active_schedules) do
+        now = Time.now()
+        enqueue_schedules(%State{state | schedules: active_schedules}, now)
```

Comment:
could we pass active_schedules as different params instead of merging it into state, because state everywhere else holds the full list and this won't be apparent when reading the enqueue_schedules code